session.go

```diff
@@ -284,8 +286,22 @@ func (s *Session) AcceptStream() (*Stream, error) {
 }
 
 // Close is used to close the session and all streams.
-// Attempts to send a GoAway before closing the connection.
+// Attempts to send a GoAway before closing the connection. The GoAway may not actually be sent depending on the
+// semantics of the underlying net.Conn. For TCP connections, it may be dropped depending on LINGER value or
+// if there's unread data in the kernel receive buffer.
```

Comment:
Why send an error in this case? I'm concerned that we're changing the semantics to block for up to 15 seconds.

Reply:
We don't need to send this on Close. Fixed this.